safehttp/plugins/angular_xsrf/angular_xsrf_test.go

```diff
@@ -0,0 +1,147 @@
+	"testing"
+)
+
+const (
+	cookieName = "XSRF-TOKEN"
+	headerName = "X-XSRF-TOKEN"
+)
+
+func TestAngularAddCookie(t *testing.T) {
+	req := safehttptest.NewRequest(safehttp.MethodGet, "/", nil)
+	rec := safehttptest.NewResponseRecorder()
+	i := &Interceptor{
+		TokenCookieName: cookieName,
+		TokenHeaderName: headerName,
+	}
+	i.Commit(rec.ResponseWriter, req, nil, nil)
+
+	if got, want := rec.Status(), safehttp.StatusOK; got != want {
+		t.Errorf("rec.Status(): got %v, want %v", got, want)
+	}
+
+	tokCookieDefaults := "Path=/; Max-Age=86400; Secure; SameSite=Strict"
+	got := map[string][]string(rec.Header())["Set-Cookie"][0]
+	if !strings.Contains(got, i.TokenCookieName) {
+		t.Errorf("Set-Cookie header: %s not present", i.TokenCookieName)
+	}
+	if !strings.Contains(got, tokCookieDefaults) {
+		t.Errorf("Set-Cookie header: got %s, want defaults %s", got, tokCookieDefaults)
+	}
+
+	if got, want := rec.Body(), ""; got != want {
+		t.Errorf("rec.Body(): got %q want %q", got, want)
+	}
+}
+
+func TestAngularPostProtection(t *testing.T) {
+	tests := []struct {
+		name       string
+		req        *safehttp.IncomingRequest
+		wantStatus safehttp.StatusCode
+		wantHeader map[string][]string
+		wantBody   string
+	}{
+		{
+			name: "Same cookie and header",
+			req: func() *safehttp.IncomingRequest {
+				req := safehttptest.NewRequest(safehttp.MethodPost, "/", nil)
+				req.Header.Set("Cookie", cookieName+"="+"1234")
+				req.Header.Set(headerName, "1234")
+				return req
+			}(),
+			wantStatus: safehttp.StatusOK,
+			wantHeader: map[string][]string{},
+			wantBody:   "",
+		},
+		{
+			name: "Different cookie and header",
+			req: func() *safehttp.IncomingRequest {
+				req := safehttptest.NewRequest(safehttp.MethodPost, "/", nil)
+				req.Header.Set("Cookie", cookieName+"="+"5768")
+				req.Header.Set(headerName, "1234")
+				return req
+			}(),
+			wantStatus: safehttp.StatusUnauthorized,
+			wantHeader: map[string][]string{
+				"Content-Type":           {"text/plain; charset=utf-8"},
+				"X-Content-Type-Options": {"nosniff"},
+			},
+			wantBody: "Unauthorized\n",
+		},
+		{
+			name: "Missing header",
+			req: func() *safehttp.IncomingRequest {
+				req := safehttptest.NewRequest(safehttp.MethodPost, "/", nil)
+				req.Header.Set("Cookie", cookieName+"="+"1234")
+				return req
+			}(),
+			wantStatus: safehttp.StatusUnauthorized,
+			wantHeader: map[string][]string{
+				"Content-Type":           {"text/plain; charset=utf-8"},
+				"X-Content-Type-Options": {"nosniff"},
+			},
+			wantBody: "Unauthorized\n",
+		},
+		{
+			name: "Missing cookie",
+			req: func() *safehttp.IncomingRequest {
+				req := safehttptest.NewRequest(safehttp.MethodPost, "/", nil)
+				req.Header.Set(headerName, "1234")
+				return req
+			}(),
+			wantStatus: safehttp.StatusForbidden,
+			wantHeader: map[string][]string{
+				"Content-Type":           {"text/plain; charset=utf-8"},
+				"X-Content-Type-Options": {"nosniff"},
+			},
+			wantBody: "Forbidden\n",
```

Comment:
These assertions feel like we are testing the framework rather than just the plugin. I would personally just test for status here. If you want to be thorough maybe also body, but the header check I think should only be a "I added a cookie or not", which would allow you to merge this test with the one above.

Reply:
Right, but we tested for the headers everywhere and I mostly did it for uniformity across tests. I suggest removing these checks, if you think they are not needed, in a follow-up PR.